src/sentry/tasks/assemble.py

```diff
@@ -164,12 +168,18 @@ def _get_cache_key(task, scope, checksum):
             % (
                 str(scope).encode("ascii"),
                 checksum.encode("ascii"),
-                str(task).encode("utf-8"),
+                str(task).encode(),
             )
         ).hexdigest()
     )
 
 
+def _get_redis_cluster_for_assemble() -> RedisCluster:
+    cluster_key = settings.SENTRY_ASSEMBLE_CLUSTER
+    return redis.redis_clusters.get(cluster_key)  # type: ignore[return-value]
```

Comment:
what's the ignore here? this should be typesafe I imagine (maybe it's expecting `RedisCluster[str]` ? or maybe we need a narrowing assertion that this isn't a non-Cluster?)

Reply:
wouldn't the normal usage of the return value break self hosted as well? should this preserve the union?

Reply:
Right now, it doesn't break because Cluster and Redis client has the same exact API surface. It will break this and many other places when we upgrade redis and remove py-cluster and rb. 

I don't know which scenario is better.

Reply:
right -- then I think this should `-> RedisCluster | rb.Cluster` instead of `# type: ignore`

Reply:
But technically this can be `StrictRedis` as well :-)


And on `utils/redis.py` we only support `RedisCluster` and `StrictRedis`.

```
self._clusters_bytes: dict[str, RedisCluster[bytes] | StrictRedis[bytes]] = {}
```


Which of them is correct?

![image](https://github.com/getsentry/sentry/assets/1935246/97ee61fc-3e4e-40a8-8379-3fbe2690cee1)